library/modules/Gui.cpp

```diff
@@ -1360,6 +1362,137 @@ DFHACK_EXPORT void Gui::writeToGamelog(std::string message)
     fseed.close();
 }
 
+bool Gui::parseReportString(std::vector<std::string> &out, const std::string &str, size_t line_length)
```

Comment:
Could this function re-use `word_wrap()` in https://github.com/DFHack/dfhack/blob/develop/library/MiscUtils.cpp#L171 ?
Like, could we pre-process the escape chars and send the strings, line by line, to `word_wrap()` for the actual wrapping?

Reply:
I tried it and ran into an issue. Testing with this string:
```
"The quick brown fox jumps over                                                   the lazy dog!"
```
DF's code gives this result:
```
"The quick brown fox jumps over                                           "
"the lazy dog!"
```
Parsing then using `word_wrap` with `ignore_space` as `false`:
```
"The quick brown fox jumps over                                           "
"       the lazy dog!"
```
Parsing then using `word_wrap` with `ignore_space` as `true`:
```
"The quick brown fox jumps over the lazy dog!"
```
Not sure if I should just accept the `true` result or manually trim space from the output strings with `false` result.

Reply:
@softmoth is this behavior from word_wrap expected? @Bumber64 how does the output differ from what you need/were expecting?

Reply:
Exceeding `line_length` in characters (including spaces) should start a new line, ignoring only the spaces on the new line until a non-space character is reached. I'm okay with losing the spaces at the end of the previous line (my "fixed" code ended up doing this as a side-effect,) but spaces should never be removed between words on the same line, and the removal of spaces should not prevent the creation of a new line.

Reply:
I wasn't trying to match any other word wrapper, and nothing in DFHack depends on its precise behavior, so I'd recommend we make word_wrap() behave as Bumber64 needs, probably. I was thinking of a *possible* case of right-aligned tabulated data which should continue to line up properly after word wrapping; thus I preserved whitespace at the beginning of the new line. But I'm not sure that this is actually needed.

Reply:
By the way, the `compress_spaces = true` result matches the previous `word_wrap()` behavior, and the separate lua function as well. Both just comb through the input looking for chunks of non-whitespace and paste them together with a single space or newline, with no regard for what the input whitespace is.

The problem with trimming the leading whitespace after `word_wrap()` is that it leaves extra space at the end of the line then. If you never have a line wrap more than once, it doesn't cause a problem, but that's an ugly assumption.

Reply:
> I was thinking of a _possible_ case of right-aligned tabulated data which should continue to line up properly after word wrapping; thus I preserved whitespace at the beginning of the new line. But I'm not sure that this is actually needed.

That's easy enough to keep by not trimming whitespace at all, right?

I just need an option for `word_wrap()` where a bool is set after pushing anything to `out`, which triggers a continue in the `if (isspace(c))` block when true, and is unset in the `else` block.

Reply:
How does this look?
https://github.com/DFHack/dfhack/compare/develop...Bumber64:patch-1